internal/services/issuer/verifiable_credential.go

```diff
@@ -180,3 +188,92 @@ func (i *Issuer) VIN(vin string, tokenID *big.Int) (id string, err error) {
 
 	return id, tx.Commit()
 }
+
+func (i *Issuer) Fingerprint(ctx goka.Context, msg interface{}) {
+	event, ok := msg.(*FingerprintEvent)
+	if !ok {
+		i.log.Err(errors.New("unable to parse fingerprint event")).Msg("invalid payload")
+		return
+	}
+	currentIssuanceWeek := GetWeekNumForCron(time.Now())
+	observedVIN, err := services.ExtractVIN(event.Data)
+	if err != nil {
+		i.log.Info().Err(err).Str("userDeviceId", event.Subject).Msg("could not extract vin from payload")
+		return
+	}
+
+	device, err := models.VehicleNFTS(
+		models.VehicleNFTWhere.UserDeviceID.EQ(null.StringFrom(event.ID)),
+		models.VehicleNFTWhere.OwnerAddress.EQ(null.BytesFrom(common.FromHex(event.Subject))),
+		qm.Load(models.VehicleNFTRels.UserDevice),
+	).One(ctx.Context(), i.DBS.DBS().Reader)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			i.log.Info().Err(err).Str("userDeviceId", event.Subject).Msg("no associated vehicle nft for device")
+			return
+		}
+		i.log.Info().Err(err).Str("userDeviceId", event.Subject).Msg("database failure retrieving user device")
+		return
+	}
+
+	tkn, ok := device.TokenID.Int64()
```

Comment:
We seem to convert this right back to `*big.Int` whenever we do anything with it.